designs/2022-supress-ignored-file-warnings/README.md

```diff
@@ -0,0 +1,157 @@
+
+For example, with
+
+```
+// .eslintignore
+foo.js
+```
+
+Running
+
+```
+eslint foo.js
+```
+
+Will result in
+
+```
+warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override
+```
+
+It is possible to filter out ignored files using ESLint `CLIEngine`. This was the main reasoning to reject the CLI flag idea during [TSC meeting in 2018](https://gitter.im/eslint/tsc-meetings/archives/2018/08/02). The commitee was only considering the use case of integrations. However, this problem affects end users as well. Lint-staged FAQ includes a section [how can I ignore files from .eslintignore](https://github.com/okonet/lint-staged#how-can-i-ignore-files-from-eslintignore), which suggests adding additional config file and using `CLIEngine.isPathIgnored` to filter out ignored files. Having a CLI flag for this would improve the end-user experience.
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+ESLint CLI should have an optional flag to disable the ignored file warning. The flag should be boolean. In order to stay consistent with current CLI flags, it should be called `--no-warning-on-ignored-files`. If it is present, ESLint should supress the `File ignored because of a matching ignore pattern.` warning. The flag should replicate `eslint.lintText`[warnIgnored option](https://eslint.org/docs/developer-guide/nodejs-api#-eslintlinttextcode-options) when it is set to `false`.
+
+The current warning should be updated to `File ignored because of a matching ignore pattern. Use "--no-warning-on-ignored-files to suppress this warning.`
+
+## Documentation
+
+<!--
+    How will this RFC be documented? Does it need a formal announcement
+    on the ESLint blog to explain the motivation?
+-->
+
+The [CLI Documentation](https://eslint.org/docs/user-guide/command-line-interface) should be updated to include the new flag.
+
+## Drawbacks
+
+<!--
+    Why should we *not* do this? Consider why adding this into ESLint
+    might not benefit the project or the community. Attempt to think
+    about any opposing viewpoints that reviewers might bring up.
+
+    Any change has potential downsides, including increased maintenance
+    burden, incompatibility with other tools, breaking existing user
+    experience, etc. Try to identify as many potential problems with
+    implementing this RFC as possible.
+-->
+
+Adding additional CLI flags adds additional mental overhead for the users. It might be confusing for users who don't experience this problem.
+
+There will be multiple was of achieving the same goal. One with `CLIEngine`, and another with the flag.
```

Comment:
Stating the obvious here in case anyone finds this alarming: the command-line argument would presumably wrap the existing `CLIEngine` method, so there would be no real duplication here, merely the same feature plumbed through an additional layer of abstraction to increase its utility.

Reply:
Yes, you are right, we're just exposing the internal option via the CLI